VirusTotal.NET/VirusTotal.cs

```diff
@@ -164,12 +178,34 @@ public ScanResult ScanFile(Stream fileStream, string filename)
             if (string.IsNullOrWhiteSpace(filename))
                 throw new ArgumentException("You must provide a filename. Preferably the original filename.");
 
-            //https://www.virustotal.com/vtapi/v2/file/scan
-            RestRequest request = PrepareRequest("file/scan");
-            request.AddFile("file", fileStream.CopyTo, filename);
+            if (fileStream.Length > PUBLIC_FILE_SIZE_UPLOAD_LIMIT)
+            {
+                //assume we are getting a private upload url
+                //https://www.virustotal.com/vtapi/v2/file/scan/upload_url
+                RestRequest uploadRequest = PrepareRequest("file/scan/upload_url", Method.GET);
```

Comment:
This is going to incur a redirect, which will fail if the user hasn't set the client to allow for redirects. Perhaps in this section of code we can set it automatically and then turn it back off after.